### DISCUSSION TITLE
Bugfix Release 1.1.1

### DISCUSSION THREAD

**Discussion by MatrixCrawler** [warrensbox/terraform-switcher#404] [Releases]
Should we aim for a bugfix release on short notice? There are issues with the version.tf parsing.
@warrensbox

**Comment by yermulnik** [comment]
I think we should resolve #401 as a primary priority and, once done, cut a bugfix release.

The other ones like #400 and #402 are more of a edge cases that fall apart the https://developer.hashicorp.com/terraform/language/expressions/version-constraints (which, I do agree, doesn't require constraint operator to be separated from version number though).

Also need to properly fix `install.sh` — I'll try to look into this though.

**Reply by yermulnik** [reply]
I'm also very keen to figure this https://github.com/warrensbox/terraform-switcher/pull/356#issuecomment-2077683539 out (@MatrixCrawler if you have time please take a look)

**Reply by MatrixCrawler** [reply]
It SHOULD work.


           if viperParser.Get("log-level") != nil {
                        params.LogLevel = viperParser.GetString("log-level")
                }

I must have a look as to why it is not applied correctly.

**Reply by yermulnik** [reply]
> Also need to properly fix `install.sh` — I'll try to look into this though.

https://github.com/warrensbox/terraform-switcher/pull/405

**Reply by MatrixCrawler** [reply]
Found the error regarding the logging.
The logger is initialized BEFORE the config files are parsed. I will add a new initialization after the config files.
See #406

**Reply by yermulnik** [reply]
@MatrixCrawler I'm still a bit confused — is it me or still `tfswitch`? 😕 
```shell
> fgrep log ~/.tfswitch.toml
log-level = "DEBUG"

> tfswitch --version
18:55:57.119 INFO Reading configuration from "/home/giermulnik/.tfswitch.toml"
Version: v1.1.1

> tfswitch --version -g DEBUG
18:55:59.475 INFO [toml.go:15,getParamsTOML] Reading configuration from "/home/giermulnik/.tfswitch.toml"
Version: v1.1.1
```

**Reply by yermulnik** [reply]
@MatrixCrawler I created an issue for this to not lose the track: https://github.com/warrensbox/terraform-switcher/issues/421

**Comment by warrensbox** [comment]
The priority is:
https://github.com/warrensbox/terraform-switcher/issues/401
https://github.com/warrensbox/terraform-switcher/pull/405

Lower priority:
https://github.com/warrensbox/terraform-switcher/issues/398

**Comment by warrensbox** [comment]
@MatrixCrawler Are you ready to merge this?
Fix terragrunt parsing #409

Let me know if you want to run Release (Manual Step)-> patch
Or I can do it.

**Reply by MatrixCrawler** [reply]
@yermulnik could you please update the changelog? Then i will try to make a patch release.

**Reply by warrensbox** [reply]
If he does not respond, I can do it.

**Reply by MatrixCrawler** [reply]
Is there a documentation on how to do it? Then i can do it myself, too. Plese feel free to do it @warrensbox

**Reply by warrensbox** [reply]
No worries. I am doing it right now

**Reply by warrensbox** [reply]
Let me know if it sounds okay?
https://github.com/warrensbox/terraform-switcher/pull/415/files

**Reply by yermulnik** [reply]
It's still semi-manual process.
So I use https://github.com/chelnak/gh-changelog like `gh changelog new --from-version 1.0.2 --next-version v1.1.0`. It creates `CHANGELOG.md.new` with the set of changes between the two tags. Then I manually copy necessary section from `CHANGELOG.md.new` to `CHANGELOG.md` and adjust it if it looks odd or broken.
Still had no time to search for something good. The `gh-changelog` does better than others I've tried so far:
* https://github.com/orhun/git-cliff
* https://github.com/git-chglog/git-chglog
* https://github.com/rafinskipg/git-changelog
* https://github.com/clog-tool/clog-cli

**Comment by yermulnik** [comment]
ℹ️ I'm closing this discussion since `v1.1.1` has been released.
